app/screens/settings/advanced_settings/advanced_settings.js

```diff
@@ -38,10 +52,65 @@ class AdvancedSettings extends PureComponent {
                 onPress: () => true
             }]
         );
+    });
+
+    clearDownloadCache = wrapWithPreventDoubleTap(() => {
+        const {intl} = this.props;
+        const {cacheSize} = this.state;
+
+        if (cacheSize !== '0 B') {
+            Alert.alert(
+                intl.formatMessage({
+                    id: 'mobile.advanced_settings.delete_file_cache',
+                    defaultMessage: 'Delete File Cache'
+                }),
+                intl.formatMessage({
+                    id: 'mobile.advanced_settings.delete_file_cache_message',
+                    defaultMessage: '\nThis will delete all the files stored in the cache. Are you sure you want to delete them?\n'
+                }),
+                [{
+                    text: intl.formatMessage({id: 'mobile.advanced_settings.delete', defaultMessage: 'Delete'}),
+                    onPress: () => {
+                        this.setState({cacheSize: null, cacheSizedFetched: false}, async () => {
+                            await deleteFileCache();
+                            this.setState({cacheSize: getFormattedFileSize({size: 0}), cacheSizedFetched: true});
+                        });
+                    }
+                }, {
+                    text: intl.formatMessage({id: 'channel_modal.cancel', defaultMessage: 'Cancel'}),
+                    onPress: () => true
+                }]
+            );
+        }
+    });
+
+    getDownloadCacheSize = async () => {
+        const size = await getFileCacheSize();
+        this.setState({cacheSize: getFormattedFileSize({size}), cacheSizedFetched: true});
```

Comment:
Why not format the cacheSize only when it's rendered the UI? This seems like it'll be error prone since I'd expect it to be a number